isisd/isis_sr.c

```diff
@@ -627,6 +627,71 @@ static int sr_local_block_release_label(struct sr_local_block *srlb,
 	return 0;
 }
 
+static bool sr_adj_same_subnet_ipv4(struct in_addr ipv4,
+				    struct isis_circuit *circuit)
+{
+	struct listnode *node;
+	struct prefix_ipv4 *ipv4_adj, *ipv4_circuit;
+	bool is_same = false;
+
+	ipv4_adj = prefix_ipv4_new();
```

Comment:
There is absolutely no need to malloc memory here.  Please just use the stack.

Reply:
ok